### PR TITLE
fix(spaces): Allow ctrl+c while running up space init

### PR DESCRIPTION
**Pull request by cbuto**
### Description of your changes

Allows using ctrl+c to interrupt the `up space init` command to allow stopping the installation without needing to kill the process via `kill -9`. 

Note: this does not allow for re-initialization of the space, killing the process or using ctrl+c could result in the space being in a unstable state. It'll be a bit of work to make the `up space init` rerunnable/idempotent. 

Fixes https://github.com/upbound/mxe/issues/257

I have:

- [x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### How has this code been tested

manually running `up space init` then using ctrl+c to exit the process while prereqs are installing.

### CODE REVIEW

File: cmd/up/space/init.go [hunk 1]
```diff
@@ -211,6 +213,14 @@ func (c *initCmd) AfterApply(kongCtx *kong.Context, quiet config.QuietFlag) erro
...
 	ctx := context.Background()
 
+	sigCh := make(chan os.Signal, 1)
+	signal.Notify(sigCh, os.Interrupt)
+	defer signal.Stop(sigCh)
+	go func() {
+		<-sigCh
+		os.Exit(1)
+	}()
+
```
Comment (sttts) [inline cmd/up/space/init.go:223]: only question: shoudn't we move that to the top-level for all commands?

File: cmd/up/main.go [hunk 2]
```diff
@@ -173,7 +175,19 @@ func main() {
...
+	sigCh := make(chan os.Signal, 1)
+	signal.Notify(sigCh, os.Interrupt)
+	defer signal.Stop(sigCh)
+	go func() {
+		defer cancel()
+		<-sigCh
+		kongCtx.Exit(1)
+	}()
+
+	kongCtx.BindTo(ctx, (*context.Context)(nil))
```
Comment (sttts) [inline cmd/up/main.go:191]: now this must be wired into all `Run` funcs where we create a Golang context.

Comment (cbuto) [reply cmd/up/main.go:191]: yup done in https://github.com/upbound/up/pull/405/commits/2d718bf0a8e21d5b2ec79dd8d97fb43c468b7d60, going to be quite the large diff but hopefully easy to review

**Review by sttts — APPROVED** [review]

**Comment by sttts** [comment]
Looks great! 🚢 it.

**Pull request merged**